From 426d9e8c2361661dc4bec62f3b700d920b83d9b7 Mon Sep 17 00:00:00 2001
From: Masahisa Kojima <masahisa.kojima@linaro.org>
Date: Tue, 16 Nov 2021 15:39:36 +0900
Subject: [PATCH 1/3] synquacer: add fTPM support

This commit adds the fTPM node and required U-Boot configuration.
fTPM can be discovered on the optee bus from linux, so fTPM node
needs to be removed when U-Boot boots linux kernel.

Signed-off-by: Masahisa Kojima <masahisa.kojima@linaro.org>
---
 .../synquacer-sc2a11-developerbox-u-boot.dtsi |  5 +++++
 arch/arm/dts/synquacer-sc2a11.dtsi            |  1 -
 board/socionext/developerbox/developerbox.c   |  3 +++
 configs/synquacer_developerbox_defconfig      | 21 +++++++++++++++++++
 4 files changed, 29 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/synquacer-sc2a11-developerbox-u-boot.dtsi b/arch/arm/dts/synquacer-sc2a11-developerbox-u-boot.dtsi
index 7a56116d6f..54a01f50db 100644
--- a/arch/arm/dts/synquacer-sc2a11-developerbox-u-boot.dtsi
+++ b/arch/arm/dts/synquacer-sc2a11-developerbox-u-boot.dtsi
@@ -7,6 +7,7 @@
 	aliases {
 		spi_nor = &spi_nor;
 		i2c0 = &i2c0;
+		ftpm = &ftpm;
 	};
 
 	spi_nor: spi@54800000 {
@@ -105,6 +106,10 @@
 			status = "okay";
 		};
 	};
+
+	ftpm: ftpm@0 {
+		compatible = "microsoft,ftpm";
+	};
 };
 
 &smmu {
diff --git a/arch/arm/dts/synquacer-sc2a11.dtsi b/arch/arm/dts/synquacer-sc2a11.dtsi
index 1fe7d214b9..460d1a575d 100644
--- a/arch/arm/dts/synquacer-sc2a11.dtsi
+++ b/arch/arm/dts/synquacer-sc2a11.dtsi
@@ -587,7 +587,6 @@
         optee {
             compatible = "linaro,optee-tz";
             method = "smc";
-            status = "disabled";
         };
     };
 };
diff --git a/board/socionext/developerbox/developerbox.c b/board/socionext/developerbox/developerbox.c
index 9552bfcdc3..401baf34b3 100644
--- a/board/socionext/developerbox/developerbox.c
+++ b/board/socionext/developerbox/developerbox.c
@@ -95,6 +95,9 @@ int ft_board_setup(void *blob, struct bd_info *bd)
 	fdt_del_node_and_alias(blob, "spi_nor");
 	fdt_del_node_and_alias(blob, "i2c0");
 
+	/* Remove fTPM, TA can be discovered on the optee bus from linux */
+	fdt_del_node_and_alias(blob, "ftpm");
+
 	return 0;
 }
 
diff --git a/configs/synquacer_developerbox_defconfig b/configs/synquacer_developerbox_defconfig
index 13736a4f03..429576c146 100644
--- a/configs/synquacer_developerbox_defconfig
+++ b/configs/synquacer_developerbox_defconfig
@@ -93,3 +93,24 @@ CONFIG_EFI_CAPSULE_ON_DISK=y
 CONFIG_EFI_IGNORE_OSINDICATIONS=y
 CONFIG_EFI_CAPSULE_FIRMWARE_RAW=y
 CONFIG_EFI_SECURE_BOOT=y
+CONFIG_TEE=y
+CONFIG_CMD_TPM_V2=y
+CONFIG_CMD_TPM=y
+CONFIG_TPM_V2=y
+CONFIG_TPM2_FTPM_TEE=y
+CONFIG_TPM=y
+CONFIG_CMD_MMC_RPMB=y
+CONFIG_CMD_OPTEE_RPMB=y
+CONFIG_SUPPORT_EMMC_RPMB=y
+CONFIG_OPTEE=y
+CONFIG_OPTEE_TA_AVB=y
+CONFIG_CHIMP_OPTEE=y
+CONFIG_OPTEE_LOAD_ADDR=0x00000000fc000000
+CONFIG_OPTEE_TZDRAM_SIZE=0x00000003c00000
+CONFIG_OPTEE_TZDRAM_BASE=0x00000000fc000000
+CONFIG_SHA1=y
+CONFIG_SHA256=y
+CONFIG_SHA512_ALGO=y
+CONFIG_SHA512=y
+CONFIG_SHA384=y
+CONFIG_MD5=y
-- 
2.17.1

